docs/source/reservations/reservations.rst

```diff
@@ -0,0 +1,182 @@
+* ``gcp.specific_reservations``: a list of reservation ids that can be used by SkyPilot. This is useful if you have multiple `specific reservations <https://cloud.google.com/compute/docs/instances/reservations-consume#consuming_instances_from_a_specific_reservation>`__ for different instance types in multiple regions/zones.
+
+.. code-block:: yaml
+
+    gcp:
+      prioritize_reservations: true
+      specific_reservations:
+        - projects/my-project/reservations/my-reservation1
+        - projects/my-project/reservations/my-reservation2
+
+
+SkyPilot will utilize the reservations similar to AWS reservations as described in :ref:`utilizing-reservations`.
+
+
+GCP Dynamic Workload Scheduling (DWS)
+-------------------------------------
+
+GCP `Dynamic Workload Scheduling (DWS) <https://cloud.google.com/blog/products/compute/introducing-dynamic-workload-scheduler>`__ is a resource management service that allows you submit a GPU capacity request, automatically provisions the requested resources when they are available and keeps the resources running for a specified duration.
+
+.. hint::
+
+    It has been observed that using DWS can significantly increase the chance of getting a high-end GPU resource, such as A100s and H100s, compared to using on-demand or spot instances.
+
+
+Using DWS for VMs
+~~~~~~~~~~~~~~~~~
+
+SkyPilot allows you to launch a cluster with DWS by specifying the ``gcp.managed_instance_group`` field in ``~/.sky/config.yaml``:
+
+.. code-block:: yaml
+
+    gcp:
+      managed_instance_group:
+        run_duration: 3600
+        provision_timeout: 900
+
+The ``run_duration`` is the duration for a created instance to be kept alive (in seconds, required), and the ``provision_timeout`` is the timeout for provisioning an instance with DWS (in seconds, optional). If the timeout is reached without getting the requested resources, SkyPilot will automatically :ref:`failover <auto-failover>` to other clouds/regions/zones to get the resources.
+
+See the :ref:`config-yaml` for more details.
+
+In case you want to specify the DWS configuration for each job/cluster, you can also specify the configuration in the SkyPilot task YAML (see :ref:`here <task-yaml-experimental>`):
+
+.. code-block:: yaml
+    experimental:
+      config_overrides:
+        gcp:
+          managed_instance_group:
+            run_duration: 3600
+            provision_timeout: 900
+
+    resources:
+      cloud: gcp
+      accelerators: A100:8
+    
+    num_nodes: 4
+    
+Using DWS on GKE with Kueue
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+DWS is also supported on Google Kubernetes Engine (GKE) with Kueue. To enable DWS on GKE, you need to set up your GKE cluster with Kueue and DWS, see the `GCP official docs <https://cloud.google.com/kubernetes-engine/docs/how-to/provisioningrequest>`
```

Comment:
fix url

Reply:
I think this is the correct doc for DWS on GKE with Kueue?